src/amberscript/parser.cc

```diff
@@ -949,6 +951,64 @@ Result Parser::ParsePipelineFramebufferSize(Pipeline* pipeline) {
   return ValidateEndOfStatement("FRAMEBUFFER_SIZE command");
 }
 
+Result Parser::ParsePipelineViewport(Pipeline* pipeline) {
+  float vp[6] = {0, 0, 0, 0, 0, 1};
```

Comment:
I agree that a struct would be nicer, but I didn't do it that way because everything else in PipelineData was individual data members. I didn't want to put in individual setters because that creates a whole lot of invalid states where one viewport component has been set but the others have not, which I'd rather avoid.